src/wallet/spend.cpp

```diff
@@ -517,40 +550,39 @@ std::optional<SelectionResult> AttemptSelection(const CWallet& wallet, const CAm
     // If we can't fund the transaction from any individual OutputType, run coin selection one last time
     // over all available coins, which would allow mixing
     if (allow_mixed_output_types) {
-        if (auto result{ChooseSelectionResult(wallet, nTargetValue, eligibility_filter, available_coins.All(), coin_selection_params)}) {
+        if (auto result{ChooseSelectionResult(wallet, nTargetValue, groups.all_groups, coin_selection_params)}) {
             return result;
         }
     }
     // Either mixing is not allowed and we couldn't find a solution from any single OutputType, or mixing was allowed and we still couldn't
     // find a solution using all available coins
-    return std::nullopt;
+    return util::Error();
 };
 
-std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, const CAmount& nTargetValue, const CoinEligibilityFilter& eligibility_filter, const std::vector<COutput>& available_coins, const CoinSelectionParams& coin_selection_params)
+util::Result<SelectionResult> ChooseSelectionResult(const CWallet& wallet, const CAmount& nTargetValue, Groups& groups,
```

Comment:
wallet is unused

Reply:
good eye, this let me remove the `AttemptSelection` wallet arg too.